xbmc/guilib/guiinfo/MusicGUIInfo.cpp

```diff
@@ -276,6 +281,16 @@ bool CMusicGUIInfo::GetLabel(std::string& value, const CFileItem *item, int cont
           return true;
         }
         break;
+      case MUSICPLAYER_STATIONNAME:
+      {
+        std::string station = tag->GetStationName();
+        // if this is a shoutcast v2 stream we have to get the station name from the sortlabel
+        // as it isn't set in the headers.
+        if (station == "shoutcast" && g_application.CurrentFileItem().IsInternetStream())
+          g_charsetConverter.wToUTF8(item->GetSortLabel(), station);
```

Comment:
Hmm, why can't the shoutcast-specific code in ShoutcastFile.cpp, line 80 calling SetStationName() not do this? Would also solve the "shoutcast" as magic title issue and make this implementation generic.

Reply:
Because the filesystem has absolutely no knowledge of the item labels.

Reply:
Fair. Why can't this be done always if GetStationName() returns nothing?

Reply:
See my comment above.  Any other type of stream will return an empty string here.  During testing I have encountered several stations via the rad.io addon that aren't using shoutcast in which case the GUI defaults back to showing the "Now Playing..." label in the top left rather than a station name.  If I only test for an empty string here, those stations would be assigned some sort of value from the sort label.  It's totally possible that it might be correct but equally possible it isn't. As this is meant to only apply to shoutcast streams the check is there to ensure that that is indeed the case.